src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -8,30 +8,39 @@ import javax.naming.NameNotFoundException
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.commons.codec.binary.Hex
 import org.broadinstitute.dsde.workbench.model._
-import org.broadinstitute.dsde.workbench.sam.api.CreateWorkbenchUser
+import org.broadinstitute.dsde.workbench.sam.api.{CreateWorkbenchUser, InviteUser}
 import org.broadinstitute.dsde.workbench.sam.directory.DirectoryDAO
 import org.broadinstitute.dsde.workbench.sam.model._
-
 import scala.concurrent.{ExecutionContext, Future}
 
 /**
   * Created by dvoet on 7/14/17.
   */
 class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExtensions)(implicit val executionContext: ExecutionContext) extends LazyLogging {
 
-  def createUser(user: CreateWorkbenchUser): Future[UserStatus] = {
+  def createUser(user: CreateWorkbenchUser): Future[UserStatus] = for {
+      createdUser <- registerUser(user)
+      res <- provisionUser(createdUser)
+    } yield res
+
+  def inviteUser(invitee: InviteUser): Future[UserStatusDetails] = for {
+      existingSubject <- directoryDAO.loadSubjectFromEmail(invitee.inviteeEmail)
+      createdUser <- existingSubject match{
+        case None => directoryDAO.createUser(WorkbenchUser(invitee.inviteeId, None, invitee.inviteeEmail))
+        case Some(__) => Future.failed(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"email ${invitee.inviteeEmail} already exists")))
+      }
+    } yield UserStatusDetails(createdUser.id, createdUser.email)
+
+  protected[service] def provisionUser(user: WorkbenchUser): Future[UserStatus] = {
```

Comment:
before invite, there's just `createUser` (what's in `provisionUser` is just what breaks out of `createUser`, because I thought `inviteUser` needs to do all these steps too...I can put it back into `createUser` if it's less confusing now that this code is actually not shared)...`inviteUser` corresponds to `invite` endpoint, and it creates an user entry in ldap, but there's nothing else set up for this user.